labelify/__main__.py

```diff
@@ -188,28 +192,129 @@ def file_or_dir_path(path):
         default="false",
     )
 
+    parser.add_argument(
+        "-u",
+        "--username",
+        type=str,
+        default=None,
+        dest="username",
+        help="sparql username",
+        required=False,
+    )
+
+    parser.add_argument(
+        "-p",
+        "--password",
+        type=str,
+        default=None,
+        dest="password",
+        help="sparql password",
+        required=False,
+    )
+
+    parser.add_argument(
+        "-g",
+        "--graph",
+        type=str,
+        default=None,
+        dest="graph",
+        help="named graph to query (only used if input is a sparql endpoint)",
+        required=False,
+    )
+
+    parser.add_argument(
+        "-r",
+        "--raw",
+        dest="raw",
+        action="store_true",
+        help="Only output the nodes with missing labels. one per line",
+    )
+
     args = parser.parse_args(args)
 
-    g = Graph().parse(args.input)
+    if isinstance(args.input, ParseResult):
+        g = Graph()
+        offset = 0
+        batch_size = 50000
+        n_results = batch_size
+        url = urlunparse(args.input)
+        if not args.raw:
+            print(
+                f"Loading triples from {url}, using graph: {'default' if not args.graph else args.graph}"
+            )
+            print(f"\tbatch_size: {batch_size}")
+        while n_results == batch_size:
+            sparql = SPARQLWrapper(endpoint=url, defaultGraph=args.graph)
+            sparql.setQuery(
+                """
+            construct {{
+                ?s ?p ?o .
+            }}
+            where {{
+                ?s ?p ?o .
+            }}
+            order by ?s
+            limit {}
+            offset {}
+            """.format(
+                    batch_size, offset
+                )
+            )
+            if args.username and not args.password:
+                sparql.setCredentials(user=args.username, passwd=getpass("password:"))
```

Comment:
Seems like this only sets the sparqlwrapper credentials if a username is supplied but a password is not?

Reply:
The CLI provides a way to pass in the password but it doesn't seem to be used by sparqlwrapper.

Reply:
I have applied a fix for this now. thanks for spotting it.